ckanext/xloader/cli.py

```diff
@@ -230,3 +233,108 @@ def _print_status(self):
                       res_id=job_metadata['resource_id'],
                       url=job_metadata['original_url'],
                       ))
+
+
+class MigrateTypesCommand(cli.CkanCommand):
+    '''Migrate command
+
+    Turn existing resource field types into Data Dictionary overrides.
+    This is intended to simplify migration from DataPusher to XLoader,
+    by allowing you to reuse the types that DataPusher has guessed.
+
+    Usage:
+
+        migrate_types [options] [resource-spec]
+            Add the given resources' field types to the Data Dictionary.
+
+            where resource-spec is one of:
+
+                <resource-id> - Migrate a particular resource
+
+                all - Migrate all resources (this is the default)
+
+    '''
+    summary = __doc__.split('\n')[0]
+    usage = __doc__
+    min_args = 0
+
+    def __init__(self, name):
+        super(MigrateTypesCommand, self).__init__(name)
+        self.error_occured = False
+
+        self.parser.add_option('-t', '--include-text',
+                               action='store_true', default=False,
+                               help='Add Data Dictionary overrides even for text fields')
+
+        self.parser.add_option('--force',
+                               action='store_true', default=False,
+                               help='Overwrite existing data dictionary if it exists')
+
+    def command(self):
+        self._load_config()
+        if not self.args or len(self.args) == 0 or self.args[0] == 'all':
+            self._migrate_all()
+        else:
+            self._migrate_resource(self.args[0])
+        self._handle_command_status()
+
+    def _migrate_all(self):
+        session = model.Session
+        resource_count = session.query(model.Resource).filter_by(state='active').count()
+        print "Updating {} resource(s)".format(resource_count)
+        resources_done = 0
+        for resource in session.query(model.Resource).filter_by(state='active'):
+            resources_done += 1
+            self._migrate_resource(resource.id, prefix='[{}/{}]: '.format(resources_done, resource_count))
+            if resources_done % 100 == 0:
+                print "[{}/{}] done".format(resources_done, resource_count)
+        print "[{}/{}] done".format(resources_done, resource_count)
+
+    def _migrate_resource(self, resource_id, prefix=''):
+        data_dict = h.datastore_dictionary(resource_id)
+
+        def print_status(status):
+            if self.options.verbose:
+                print "{}{}: {}".format(prefix, resource_id, status)
+
+        if not data_dict:
+            print_status("not found")
+            return
+
+        fields = []
+        for field in data_dict:
+            if field['type'] == 'text' and not self.options.include_text:
+                type_override = ''
+            else:
+                type_override = field['type']
+
+            if not 'info' in field:
```

Comment:
```suggestion
            if 'info' not in field:
```
and just some other minor flake8 errors to fix in this file, to make Travis pass

Reply:
This is done.